### PR TITLE
QAE-211 - Test case for primary menu submenu width in customizer

### PR DESCRIPTION
**Pull request by bsfprajaktap**
Description
This test case is for primary menu submenu width in customizer

Screenshots
https://share.bsf.io/xQu6wKxy

Types of changes
Testing the submenu width in desktop

How has this been tested?
Create menu
Create submenu
Navigate to header builder in customizer
Set the width of submenu

cmd to execute TC-
npm run test:e2e:interactive tests/e2e/specs/customizer/header-builder/primary-menu-submenu/width.test.js

Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: tests/e2e/utils/create-menu.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { createURL, createNewPost, publishPost } from '@wordpress/e2e-test-utils';
```
Comment (bsfsnehal) [inline tests/e2e/utils/create-menu.js:1]: "publish-post" should be imported from "utils" folder itself

File: tests/e2e/specs/customizer/header-builder/primary-menu-submenu/primary-menu-submenu-width.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		} );
+		await page.hover( '.menu-link' );
+		await expect( {
+			selector: '.sub-menu',
+			property: 'width',
+		} ).cssValueToBe(
+			`${ submenuWidth[ 'header-menu1-submenu-width' ] + 'px' }`,
+		);
+	} );
+} );
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/primary-menu-submenu/primary-menu-submenu-width.test.js:40]: unnecessary file. It should be removed

File: tests/e2e/specs/menu.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+			},
+		};
+		await setCustomize( headerMenuAlignment );
+		await page.goto( createURL( '/' ), {
+			waitUntil: 'networkidle0',
+		} );
+		await page.waitForSelector( '#primary-site-navigation' );
+		await expect( true ).toBe( true );
+	} );
+} );
```
Comment (bsfsnehal) [inline tests/e2e/specs/menu.test.js:27]: unnecessary file.

**Comment by bsfsnehal** [comment]
@SimranChougule1 please check the folder structure for this test case

**Review by bsfsnehal — APPROVED** [review]

File: tests/e2e/specs/customizer/header-builder/primary-menu-submenu/width.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { createURL } from '@wordpress/e2e-test-utils';
+import { createNewMenu } from '../../../../utils/create-menu';
+import { setCustomize } from '../../../../utils/customize';
+describe( 'Add sub menu for primary menu', () => {
+	it( 'sub menu should be added successfully', async () => {
+		await createNewMenu();
+		//center alignment for desktop, tablet and mobile
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/primary-menu-submenu/width.test.js:7]: remove comment if it is not necessary

File: tests/e2e/specs/customizer/header-builder/primary-menu-submenu/width.test.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+		const submenuWidth = {
+			'header-menu1-submenu-width': '100',
+		};
+		await setCustomize( submenuWidth );
+		await page.goto( createURL( '/' ), {
+			waitUntil: 'networkidle0',
+		} );
+		await page.hover( '.menu-link' );
+		await expect( {
+			selector: '.sub-menu',
```
[the remark this answers is not in the document]
Comment (bsfsnehal) [reply tests/e2e/specs/customizer/header-builder/primary-menu-submenu/width.test.js]: resolved

**Review by GauravKhupse — APPROVED** [review]

**Pull request merged**